### PR TITLE
Check notes list is not getting too long

### PR DESCRIPTION
**Pull request by philipbl**

### CODE REVIEW

File: src/yaudio.cpp [hunk 1]
```diff
@@ -14,6 +14,7 @@ i2s_port_t I2S_PORT = I2S_NUM_0;
 static const int BITS_PER_SAMPLE = 16;
 static const int BYTES_PER_SAMPLE = BITS_PER_SAMPLE / 8;
 static const int SAMPLE_RATE = 16000; // sample rate in Hz
+static const int MAX_NOTES_IN_BUFFER = 14000;
```
Comment (philipbl) [inline src/yaudio.cpp]: @jgoeders, I can't remember what size you thought the max size should be? I thought you said 14k?

Comment (jgoeders) [reply src/yaudio.cpp]: I was thinking 4k, but I don't have any good justification.

File: src/yaudio.cpp [hunk 2]
```diff
@@ -71,13 +72,17 @@ static void set_note_defaults();
...
-void add_notes(const std::string &new_notes) {
+bool add_notes(const std::string &new_notes) {
     // If WAVE is running, then stop it and reset the audio buffer.
     // Otherwise if notes are running, it's fine to let them keep running.
     if (wave_running) {
         stop();
     }
 
+    if ((notes.length() + new_notes.length()) > MAX_NOTES_IN_BUFFER) {
+        return false;
```
Comment (philipbl) [inline src/yaudio.cpp:85]: @jgoeders, do you have thoughts on what we should do if the size is too big? Right now I just give up and return false. We could add the remaining notes to the buffer to fill it up but then the user wouldn't know what got put into the buffer and what didn't get put in the buffer.

**Comment by jgoeders** [comment]
@philipbl Are you going to propagate this return value up to the yboard.h function?  And print an error message on failure?

File: src/yaudio.cpp (hunk 1 above)
Comment (philipbl) [reply src/yaudio.cpp]: I must have heard you wrong. 14k did seem kind of large. I will change it to 4K.

**Comment by philipbl** [comment]
Good point. I will look into that.

File: src/yaudio.cpp (hunk 2 above)
Comment (jgoeders) [reply src/yaudio.cpp:85]: I don't think buffering the remaining notes is any different that just allowing a longer string.  I think we just need to throw them away.  The current issue is that the board will crash and run out of memory if they call this function in a loop.

**Review by jgoeders — APPROVED** [review]

**Pull request merged**